api/src/pcapi/alembic/versions/20231009T092610_631df26e514b_add_last_author_validation_or_rejection_on_collective_offer_table.py

```diff
@@ -12,19 +14,29 @@
 branch_labels = None
 depends_on = None
 
+FOREIGN_KEY_NAME = "collective_offer_validation_author_fkey"
+
 
 def upgrade() -> None:
     op.add_column("collective_offer", sa.Column("lastValidationAuthorUserId", sa.BigInteger(), nullable=True))
+    op.execute("""SET SESSION statement_timeout='300s'""")
     op.create_foreign_key(
-        "collective_offer_validation_author_fkey",
+        FOREIGN_KEY_NAME,
         "collective_offer",
         "user",
         ["lastValidationAuthorUserId"],
         ["id"],
         ondelete="SET NULL",
+        postgresql_not_valid=True,
```

Comment:
Il ne faut pas une deuxième migration pour valider du coup ?

Reply:
Par ailleurs le timeout est nécessaire si on créé la contrainte en mode `NOT VALID` ? C’est pas justement la validation (des rows existantes) qui prend du temps ?

Reply:
Je retire les timeouts et je downgrade alembic sur staging pour tester ces migrations lors de la prochaine MES.
Pour la validation, comme c'est une nouvelle colonne, selon moi c'est pas obligatoire. Il validera la contrainte pour les nouvelles lignes et les lignes modifiés. J'ai testé en local, j'espère que c'est de même en staging, je testerais.

Reply:
mais si il n’y a aucune colonne existante à valider pourquoi ça atteignait le timeout je ne comprends pas du coup ?

Reply:
poke @francois-seguin tu as déjà fait ça je crois

Reply:
@damien-ramelet @francois-seguin @dbaty  J'ai retiré le timeout sur la colonne pas valide et j'ai fait des migrations pour valider les FK dans une autre migration (avec timeout) ça me semble être la manière correcte maintenant ?

Reply:
Sur staging, c’est effectivement instant en mode `NOT VALID`

```
staging> ALTER TABLE collective_offer ADD CONSTRAINT "toto" FOREIGN KEY ("lastValidationAuthorUserId") REFERENCES "user" ("id") ON DELETE SET NULL NOT VALID
You're about to run a destructive command.
Do you want to proceed? (y/n): y
Your call!
ALTER TABLE
Time: 0.043s
staging> commit
COMMIT
Time: 0.004s
staging> describe collective_offer
...
    "toto" FOREIGN KEY ("lastValidationAuthorUserId") REFERENCES "user"(id) ON DELETE SET NULL NOT VALID
```

Reply:
je suis chiant si je te demande deux commit séparé ? c’est plus facile pour relire et pour tracer les changements dans l’historique git

Reply:
sinon ça m’a l’air ok